src/App.tsx

```diff
@@ -1,39 +1,66 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type AppState = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends React.Component<{}, AppState> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerClockNameId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    document.addEventListener('click', () => {
+      if (!this.state.hasClock) {
+        this.setState({ hasClock: true });
+      }
+    });
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('contextmenu', (event: Event) => {
+      if (this.state.hasClock) {
+        event.preventDefault();
+        this.setState({ hasClock: false });
+      }
+    });
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    this.timerClockNameId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentDidUpdate(
+    _prevProps: Readonly<{}>, prevState: Readonly<AppState>,
+  ): void {
+    if (prevState.clockName !== this.state.clockName && this.state.hasClock) {
+      // eslint-disable-next-line
+      console.debug(`Renamed from ${prevState.clockName} to ${this.state.clockName}`);
+    }
+  }
```

Comment:
Maybe do it in the Clock component? You pass the name there anyway, and you would need to only use prevProps there